src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.MakeMethodAsynchronous.AbstractMakeMethodAsynchronousCodeFixProvider;
+
+namespace Microsoft.CodeAnalysis.RemoveAsyncModifier
+{
+    internal abstract class AbstractRemoveAsyncModifierCodeFixProvider : CodeFixProvider
+    {
+        public static readonly string EquivalenceKey = FeaturesResources.Remove_async_modifier;
+
+        protected abstract bool IsAsyncSupportingFunctionSyntax(SyntaxNode node);
+        protected abstract SyntaxNode RemoveAsyncModifier(IMethodSymbol methodSymbolOpt, SyntaxNode node, KnownTypes knownTypes);
+        protected abstract SyntaxNode ChangeReturnStatements(SyntaxNode node, SyntaxGenerator generator, KnownTypes knownTypes);
+        protected virtual bool CanFix(KnownTypes knownTypes, IMethodSymbol methodSymbolOpt) => true;
```

Comment:
```suggestion

        protected virtual bool CanFix(KnownTypes knownTypes, IMethodSymbol methodSymbolOpt) => true;
```

i.e. separate abstracts from virtuals.